packages/gatsby-source-contentful/src/source-nodes.js

```diff
@@ -438,17 +437,20 @@ export async function sourceNodes(
     reporter.info(`Creating ${assets.length} Contentful asset nodes`)
   }
 
+  let assetNodes = []
   for (let i = 0; i < assets.length; i++) {
     // We wait for each asset to be process until handling the next one.
-    await Promise.all(
-      createAssetNodes({
-        assetItem: assets[i],
-        createNode,
-        createNodeId,
-        defaultLocale,
-        locales,
-        space,
-      })
+    assetNodes = assetNodes.concat(
```

Comment:
assetNodes.push(...arrayOfNodes) will be better (won't create bunch of temporary arrays to be garbage collected lated)